src/migration-scripts/ipsec/9-to-10

```diff
@@ -0,0 +1,119 @@
+    exit(1)
+
+file_name = argv[1]
+
+with open(file_name, 'r') as f:
+    config_file = f.read()
+
+base = ['vpn', 'ipsec']
+config = ConfigTree(config_file)
+
+if not config.exists(base):
+    # Nothing to do
+    exit(0)
+
+# IKE changes, T4118:
+if config.exists(base + ['ike-group']):
+    for ike_group in config.list_nodes(base + ['ike-group']):
+        # replace 'ipsec ike-group <tag> mobike disable'
+        #      => 'ipsec ike-group <tag> disable-mobike'
+        mobike = base + ['ike-group', ike_group, 'mobike']
+        if config.exists(mobike):
+            if config.return_value(mobike) == 'disable':
+                config.set(base + ['ike-group', ike_group, 'disable-mobike'])
+            config.delete(mobike)
+
+        # replace 'ipsec ike-group <tag> ikev2-reauth yes'
+        #      => 'ipsec ike-group <tag> ikev2-reauth'
+        reauth = base + ['ike-group', ike_group, 'ikev2-reauth']
+        if config.exists(reauth):
+            if config.return_value(reauth) == 'yes':
+                config.delete(reauth)
+                config.set(reauth)
+            else:
+                config.delete(reauth)
+
+# ESP changes
+# replace 'ipsec esp-group <tag> compression enable'
+#      => 'ipsec esp-group <tag> enable-compression'
+if config.exists(base + ['esp-group']):
+    for esp_group in config.list_nodes(base + ['esp-group']):
+        compression = base + ['esp-group', esp_group, 'compression']
+        if config.exists(compression):
+            if config.return_value(compression) == 'enable':
+                config.set(base + ['esp-group', esp_group, 'enable-compression'])
+            config.delete(compression)
+
+# PEER changes
+if config.exists(base + ['site-to-site', 'peer']):
+    for peer in config.list_nodes(base + ['site-to-site', 'peer']):
+        # replace: 'peer <tag> id x'
+        #       => 'peer <tag> local-id x'
+        if config.exists(base + ['site-to-site', 'peer', peer, 'authentication', 'id']):
+            config.rename(base + ['site-to-site', 'peer', peer, 'authentication', 'id'], 'local-id')
+
+        # For the peer '@foo' set remote-id 'foo' if remote-id is not defined
+        if peer.startswith('@'):
+            if not config.exists(base + ['site-to-site', 'peer', peer, 'authentication', 'remote-id']):
+                tmp = peer.replace('@', '')
+                config.set(base + ['site-to-site', 'peer', peer, 'authentication', 'remote-id'], value=tmp)
+
+        # replace: 'peer <tag> force-encapsulation enable'
+        #       => 'peer <tag> force-encapsulation'
+        force_enc = base + ['site-to-site', 'peer', peer, 'force-encapsulation']
+        if config.exists(force_enc):
+            if config.return_value(force_enc) == 'enable':
+                config.delete(force_enc)
+                config.set(force_enc)
+            else:
+                config.delete(force_enc)
+
+        # add option: 'peer <tag> remote-address x.x.x.x'
+        remote_address = peer.replace('@', '')
```

Comment:
This naming is confusing. If a peer name is a _remote address_, it will never contain a `@` character. If it contains `@`, it's a remote ID, not an address.

I think the code for peer renaming doesn't really need any conditionals. Replacing `@` with `''` is idempotent and simply does nothing if there's no `@`.

Reply:
Done!
```
        remote_address = peer
        if peer.startswith('@'):
            remote_address = 'any'
```